### ISSUE TITLE
Incompatibility with `docker-ce==20.10.19-3`?

### ISSUE THREAD

**Issue by edvgui** [srl-labs/containerlab#1069]
Our internal lab started using `docker-ce==20.10.19-3` and since then, we are seeing those types of issues when trying to deploy a new lab (on a machine that already has one lab deployed)

```console
[root@clab-1 scripts]# clab deploy --topo cisco-2R.clab.yml 
INFO[0000] Containerlab v0.32.2 started                 
INFO[0000] Parsing & checking topology file: cisco-2R.clab.yml 
INFO[0000] Creating lab directory: /opt/connect-lab/scripts/clab-cisco-2R 
INFO[0000] Creating docker network: Name="cisco-2R", IPv4Subnet="172.20.20.0/24", IPv6Subnet="", MTU="1500" 
Error: Error response from daemon: failed to allocate gateway (172.20.22.1): Requested address is out of range
```

Might be interesting to note that the lab that is already deployed has the subnet `172.20.22.0/24`.

Given that the failure in our lab started with the update of docker, I would have assumed that the issue would coming from their side, the reason I am posting the issue here anyway is that when creating the network manually using docker, then starting the lab, things go as expected.

```
[root@clab-1 scripts]# docker network create --subnet 172.20.20.0/24 --label containerlab="" cisco-2R
dbb1e1a626201a7bbfcb7cb9dc9e16598b3aa226fb73ec7ddd3d674d3fe38794
[root@clab-1 scripts]# docker network inspect cisco-2R
[
    {
        "Name": "cisco-2R",
        "Id": "dbb1e1a626201a7bbfcb7cb9dc9e16598b3aa226fb73ec7ddd3d674d3fe38794",
        "Created": "2022-10-17T08:14:12.032201715+02:00",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": {},
            "Config": [
                {
                    "Subnet": "172.20.20.0/24"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {},
        "Options": {},
        "Labels": {
            "containerlab": ""
        }
    }
]
[root@clab-1 scripts]# clab deploy --topo cisco-2R.clab.yml 
INFO[0000] Containerlab v0.32.2 started                 
INFO[0000] Parsing & checking topology file: cisco-2R.clab.yml 
INFO[0000] Creating lab directory: /opt/connect-lab/scripts/clab-cisco-2R 
INFO[0000] Creating container: "inmanta-db"             
INFO[0000] Creating container: "subscriber-south-west"  
INFO[0000] Creating container: "inmanta-server"         
INFO[0000] Creating container: "subscriber-north-west"  
INFO[0000] Creating container: "subscriber-south-east"  
INFO[0000] Creating container: "subscriber-north-east"  
INFO[0000] Creating container: "router-east"            
INFO[0000] Creating container: "router-west"            
INFO[0000] Creating virtual wire: router-east:eth3 <--> subscriber-south-east:eth1 
INFO[0000] Creating virtual wire: router-east:eth2 <--> subscriber-north-east:eth1 
INFO[0000] Creating virtual wire: router-west:eth2 <--> subscriber-north-west:eth1 
INFO[0000] Creating virtual wire: router-west:eth3 <--> subscriber-south-west:eth1 
INFO[0000] Creating virtual wire: router-east:eth1 <--> router-west:eth1 
INFO[0001] Adding containerlab host entries to /etc/hosts file
```

Does someone here know what might be going on there?

**Comment by hellt** [comment]
Hi @edvgui 
does your topology file specify the management network name specifically?
Can you share the output of `docker network ls` command once you create your network cisco-2R?

It shouldn't be possible to create two networks with the same name...

**Comment by edvgui** [comment]
Hello @hellt 

I created a smaller setup to make it easier reproducing the issue, here it is:

<details>

<summary>lab1.yml</summary>

```yml
name: srl_lab_1
mgmt:
  network: net-1
  ipv4_subnet: 172.20.21.0/24
topology:
  nodes:
    srl1:
      kind: srl
      type: ixrd3
      image: ghcr.io/nokia/srlinux
```

</details>

<details>

<summary>lab2.yml</summary>

```yml
name: srl_lab_2
mgmt:
  network: net-2
  ipv4_subnet: 172.20.22.0/24
topology:
  nodes:
    srl1:
      kind: srl
      type: ixrd3
      image: ghcr.io/nokia/srlinux
```

</details>

```console
guillaume@latitude-5411:~/Desktop/clab-bug-1069$ docker network ls
NETWORK ID     NAME      DRIVER    SCOPE
aefdc0997baf   bridge    bridge    local
fd51accc683a   host      host      local
55dc87e303ca   none      null      local
guillaume@latitude-5411:~/Desktop/clab-bug-1069$ sudo clab deploy --topo lab2.yml 
[sudo] password for guillaume: 
INFO[0000] Containerlab v0.32.2 started                 
INFO[0000] Parsing & checking topology file: lab2.yml   
INFO[0000] Creating lab directory: /home/guillaume/Desktop/clab-bug-1069/clab-srl_lab_2 
INFO[0000] Creating docker network: Name="net-2", IPv4Subnet="172.20.22.0/24", IPv6Subnet="", MTU="1500" 
INFO[0000] Creating container: "srl1"                   
INFO[0000] Running postdeploy actions for Nokia SR Linux 'srl1' node 
INFO[0010] Adding containerlab host entries to /etc/hosts file 
+---+---------------------+--------------+-----------------------+------+---------+----------------+--------------+
| # |        Name         | Container ID |         Image         | Kind |  State  |  IPv4 Address  | IPv6 Address |
+---+---------------------+--------------+-----------------------+------+---------+----------------+--------------+
| 1 | clab-srl_lab_2-srl1 | 9f012de84e0f | ghcr.io/nokia/srlinux | srl  | running | 172.20.22.2/24 | N/A          |
+---+---------------------+--------------+-----------------------+------+---------+----------------+--------------+
guillaume@latitude-5411:~/Desktop/clab-bug-1069$ sudo clab deploy --topo lab1.yml 
INFO[0000] Containerlab v0.32.2 started                 
INFO[0000] Parsing & checking topology file: lab1.yml   
INFO[0000] Creating lab directory: /home/guillaume/Desktop/clab-bug-1069/clab-srl_lab_1 
INFO[0000] Creating docker network: Name="net-1", IPv4Subnet="172.20.21.0/24", IPv6Subnet="", MTU="1500" 
Error: Error response from daemon: failed to allocate gateway (172.20.22.1): Requested address is out of range
guillaume@latitude-5411:~/Desktop/clab-bug-1069$ docker network ls
NETWORK ID     NAME      DRIVER    SCOPE
aefdc0997baf   bridge    bridge    local
fd51accc683a   host      host      local
958dd51ea2f8   net-2     bridge    local
55dc87e303ca   none      null      local
guillaume@latitude-5411:~/Desktop/clab-bug-1069$ 
```

**Comment by hellt** [comment]
mm, interesting.
Now it is clear that the issue stems from the fact that the allocating gateway is indeed requested from a different network.

I will have a look

**Comment by hellt** [comment]
Hi @edvgui 
can you run this command to download a beta build where this issue is fixed and try it out?

```
docker run --rm -v $(pwd):/workspace ghcr.io/oras-project/oras:v0.12.0 pull ghcr.io/srl-labs/clab-oci:55aa184c
```

with this command a binary named `containerlab` will appear in the $PWD, you can use it as a replacement for a globally installed containerlab like
```
./containerlab <command>
```

**Comment by edvgui** [comment]
Hello @hellt ,

It looks like the fix is working for me, thanks a lot!  
I assume this will be part of the next release?

<details>

<summary>Logs</summary>

```console
guillaume@ubuntu-vm:~$ sudo ./containerlab deploy --topo lab1.yml 
INFO[0000] Containerlab v0.0.0 started                  
INFO[0000] Parsing & checking topology file: lab1.yml   
INFO[0000] Could not read docker config: open /root/.docker/config.json: no such file or directory 
INFO[0000] Pulling ghcr.io/nokia/srlinux:latest Docker image 
INFO[0028] Done pulling ghcr.io/nokia/srlinux:latest    
WARN[0028] it appears that container host has low memory available: ~0Gi. This might lead to runtime errors. Consider freeing up more memory. 
INFO[0028] Creating lab directory: /home/guillaume/clab-srl_lab_1 
INFO[0028] Creating docker network: Name="net-1", IPv4Subnet="172.20.21.0/24", IPv6Subnet="", MTU="1500" 
INFO[0028] Creating container: "srl1"                   
WARN[0028] Cannot parse export template /etc/containerlab/templates/export/auto.tmpl: open /etc/containerlab/templates/export/auto.tmpl: no such file or directory 
INFO[0028] Running postdeploy actions for Nokia SR Linux 'srl1' node 
INFO[0039] Adding containerlab host entries to /etc/hosts file 
INFO[0039] 🎉 New containerlab version 0.32.2 is available! Release notes: https://containerlab.dev/rn/0.32/#0322
Run 'containerlab version upgrade' to upgrade or go check other installation options at https://containerlab.dev/install/ 
+---+---------------------+--------------+-----------------------+------+---------+----------------+--------------+
| # |        Name         | Container ID |         Image         | Kind |  State  |  IPv4 Address  | IPv6 Address |
+---+---------------------+--------------+-----------------------+------+---------+----------------+--------------+
| 1 | clab-srl_lab_1-srl1 | a8eee8d3628f | ghcr.io/nokia/srlinux | srl  | running | 172.20.21.2/24 | N/A          |
+---+---------------------+--------------+-----------------------+------+---------+----------------+--------------+
guillaume@ubuntu-vm:~$ sudo ./containerlab deploy --topo lab2.yml 
INFO[0000] Containerlab v0.0.0 started                  
INFO[0000] Parsing & checking topology file: lab2.yml   
WARN[0000] it appears that container host has low memory available: ~0Gi. This might lead to runtime errors. Consider freeing up more memory. 
INFO[0000] Creating lab directory: /home/guillaume/clab-srl_lab_2 
INFO[0000] Creating docker network: Name="net-2", IPv4Subnet="172.20.22.0/24", IPv6Subnet="", MTU="1500" 
INFO[0000] Creating container: "srl1"                   
WARN[0000] Cannot parse export template /etc/containerlab/templates/export/auto.tmpl: open /etc/containerlab/templates/export/auto.tmpl: no such file or directory 
INFO[0000] Running postdeploy actions for Nokia SR Linux 'srl1' node 
INFO[0011] Adding containerlab host entries to /etc/hosts file 
INFO[0011] 🎉 New containerlab version 0.32.2 is available! Release notes: https://containerlab.dev/rn/0.32/#0322
Run 'containerlab version upgrade' to upgrade or go check other installation options at https://containerlab.dev/install/ 
+---+---------------------+--------------+-----------------------+------+---------+----------------+--------------+
| # |        Name         | Container ID |         Image         | Kind |  State  |  IPv4 Address  | IPv6 Address |
+---+---------------------+--------------+-----------------------+------+---------+----------------+--------------+
| 1 | clab-srl_lab_2-srl1 | c9ea70345c43 | ghcr.io/nokia/srlinux | srl  | running | 172.20.22.2/24 | N/A          |
+---+---------------------+--------------+-----------------------+------+---------+----------------+--------------+
guillaume@ubuntu-vm:~$ sudo docker network ls
NETWORK ID     NAME      DRIVER    SCOPE
265c2cc46817   bridge    bridge    local
2db58ecdd4f5   host      host      local
7cc9017ac9b8   net-1     bridge    local
745801443feb   net-2     bridge    local
d7e3bd9e1fe2   none      null      local
guillaume@ubuntu-vm:~$ sudo ./containerlab destroy --topo lab2.yml 
INFO[0000] Parsing & checking topology file: lab2.yml   
INFO[0000] Destroying lab: srl_lab_2                    
INFO[0000] Removed container: clab-srl_lab_2-srl1       
INFO[0000] Removing containerlab host entries from /etc/hosts file 
guillaume@ubuntu-vm:~$ sudo ./containerlab deploy --topo lab2.yml 
INFO[0000] Containerlab v0.0.0 started                  
INFO[0000] Parsing & checking topology file: lab2.yml   
INFO[0000] Creating lab directory: /home/guillaume/clab-srl_lab_2 
INFO[0000] Creating docker network: Name="net-2", IPv4Subnet="172.20.22.0/24", IPv6Subnet="", MTU="1500" 
INFO[0000] Creating container: "srl1"                   
WARN[0000] Cannot parse export template /etc/containerlab/templates/export/auto.tmpl: open /etc/containerlab/templates/export/auto.tmpl: no such file or directory 
INFO[0000] Running postdeploy actions for Nokia SR Linux 'srl1' node 
INFO[0009] Adding containerlab host entries to /etc/hosts file 
INFO[0009] 🎉 New containerlab version 0.32.2 is available! Release notes: https://containerlab.dev/rn/0.32/#0322
Run 'containerlab version upgrade' to upgrade or go check other installation options at https://containerlab.dev/install/ 
+---+---------------------+--------------+-----------------------+------+---------+----------------+--------------+
| # |        Name         | Container ID |         Image         | Kind |  State  |  IPv4 Address  | IPv6 Address |
+---+---------------------+--------------+-----------------------+------+---------+----------------+--------------+
| 1 | clab-srl_lab_2-srl1 | 947100265d05 | ghcr.io/nokia/srlinux | srl  | running | 172.20.22.2/24 | N/A          |
+---+---------------------+--------------+-----------------------+------+---------+----------------+--------------+
guillaume@ubuntu-vm:~$ sudo ./containerlab destroy --topo lab1.yml 
INFO[0000] Parsing & checking topology file: lab1.yml   
INFO[0000] Destroying lab: srl_lab_1                    
INFO[0000] Removed container: clab-srl_lab_1-srl1       
INFO[0000] Removing containerlab host entries from /etc/hosts file 
guillaume@ubuntu-vm:~$ sudo ./containerlab deploy --topo lab1.yml 
INFO[0000] Containerlab v0.0.0 started                  
INFO[0000] Parsing & checking topology file: lab1.yml   
INFO[0000] Creating lab directory: /home/guillaume/clab-srl_lab_1 
INFO[0000] Creating docker network: Name="net-1", IPv4Subnet="172.20.21.0/24", IPv6Subnet="", MTU="1500" 
INFO[0000] Creating container: "srl1"                   
WARN[0000] Cannot parse export template /etc/containerlab/templates/export/auto.tmpl: open /etc/containerlab/templates/export/auto.tmpl: no such file or directory 
INFO[0000] Running postdeploy actions for Nokia SR Linux 'srl1' node 
INFO[0009] Adding containerlab host entries to /etc/hosts file 
INFO[0009] 🎉 New containerlab version 0.32.2 is available! Release notes: https://containerlab.dev/rn/0.32/#0322
Run 'containerlab version upgrade' to upgrade or go check other installation options at https://containerlab.dev/install/ 
+---+---------------------+--------------+-----------------------+------+---------+----------------+--------------+
| # |        Name         | Container ID |         Image         | Kind |  State  |  IPv4 Address  | IPv6 Address |
+---+---------------------+--------------+-----------------------+------+---------+----------------+--------------+
| 1 | clab-srl_lab_1-srl1 | 3ea83fb785a7 | ghcr.io/nokia/srlinux | srl  | running | 172.20.21.2/24 | N/A          |
+---+---------------------+--------------+-----------------------+------+---------+----------------+--------------+
guillaume@ubuntu-vm:~$ sudo ./containerlab destroy --topo lab1.yml 
INFO[0000] Parsing & checking topology file: lab1.yml   
INFO[0000] Destroying lab: srl_lab_1                    
INFO[0000] Removed container: clab-srl_lab_1-srl1       
INFO[0000] Removing containerlab host entries from /etc/hosts file 
guillaume@ubuntu-vm:~$ sudo ./containerlab destroy --topo lab2.yml 
INFO[0000] Parsing & checking topology file: lab2.yml   
INFO[0000] Destroying lab: srl_lab_2                    
INFO[0000] Removed container: clab-srl_lab_2-srl1       
INFO[0000] Removing containerlab host entries from /etc/hosts file 
guillaume@ubuntu-vm:~$ sudo docker network ls
NETWORK ID     NAME      DRIVER    SCOPE
265c2cc46817   bridge    bridge    local
2db58ecdd4f5   host      host      local
d7e3bd9e1fe2   none      null      local
```

</details>

**Comment by hellt** [comment]
There are some things yet to be fixed, but when I do that I'll issue a patch release 0.32.3

**Comment by hellt** [comment]
@edvgui fixed in 0.32.3
